File: youlexuan-manager-web/src/main/webapp/js/service/typeTemplateService.js
/*
*定义服务
*/
app.service('typeTemplateService', function ($http) {

    //查询所有
    this.selectAll = function () {
        return $http.get("../typeTemplate/selectAll.do");
    };

    //查询品牌和规格
    this.selectBrandAndSpecification = function () {
        return $http.get('../typeTemplate/selectBrandAndSpecification.do');
    };

    //分页列表
    this.selectPage = function (page, rows) {
        return $http.get('../typeTemplate/selectPage.do?page=' + page + '&rows=' + rows);
    };

    //增加模板
    this.insert = function (entity) {
        return $http.post('../typeTemplate/insert.do', entity);
    };

    //修改模板
    this.update = function (entity) {
        return $http.post('../typeTemplate/update.do', entity);
    };

    //根据ID获取实体
    this.selectOne = function (id) {
        return $http.get('../typeTemplate/selectOne.do?id=' + id);
    };

    //删除模板
    this.delete = function (ids) {
        return $http.get('../typeTemplate/delete.do?ids=' + ids);
    };

    //条件查询
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../typeTemplate/search.do?pageNum=' + pageNum + '&pageSize=' + pageSize, searchEntity);
    }

})

File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/TypeTemplateService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;

/*
模板服务层接口
 */
public interface TypeTemplateService {

    /*
    返回指定模板id的规格列表，包含规格选项（根据模板id查询规格列表spec_ids）
     */
    public List<Map> findSpecList(Long id);

    /*
    查询所有
     */
    public List<TbTypeTemplate> selectAll();

    /*
    查询品牌和规格
     */
    public Map selectBrandAndSpecification();

    /*
    分页列表
     */
    public PageResult selectPage(int pageNum, int pageSize);

    /*
    增加模板
     */
    public void insert(TbTypeTemplate tbTypeTemplate);

    /*
    修改模板
     */
    public void update(TbTypeTemplate tbTypeTemplate);

    /*
    根据ID获取实体
     */
    public TbTypeTemplate selectOne(Long id);

    /*
    删除模板
     */
    public void delete(Long[] ids);

    /*
    条件查询
     */
    public PageResult findPage(int pageNum, int pageSize, TbTypeTemplate typeTemplate);

}
